specs/BrowserProcessExited.md

```diff
@@ -66,25 +78,46 @@ void AppWindow::CloseWebView(/* ... */) {
               ICoreWebView2Environment* sender,
               ICoreWebView2BrowserProcessExitedEventArgs* args) {
           COREWEBVIEW2_BROWSER_PROCESS_EXIT_KIND kind;
+          UINT32 pid;
           CHECK_FAILURE(args->get_BrowserProcessExitKind(&kind));
-
-          // Watch for graceful browser process exit. Let ProcessFailed event
-          // handler take care of failed browser process termination.
-          if (kind == COREWEBVIEW2_BROWSER_PROCESS_EXIT_KIND_NORMAL)
+          CHECK_FAILURE(args->get_BrowserProcessId(&pid));
+
+          // If a new WebView is created from this CoreWebView2Environment after
+          // the browser has exited but before our handler gets to run, a new
+          // browser process will be created and lock the user data folder
+          // again. Do not attempt to cleanup the user data folder in these
+          // cases. If this happens, the PID of the new browser process will be
+          // different to the PID of the older process, we check against the
+          // PID of the browser process to which our last CoreWebView2 attached.
+          if (pid == m_newestBrowserPid)
+          {
+            // Watch for graceful browser process exit. Let ProcessFailed event
+            // handler take care of failed browser process termination.
+            if (kind == COREWEBVIEW2_BROWSER_PROCESS_EXIT_KIND_NORMAL)
+            {
+              CHECK_FAILURE(
+                  m_webViewEnvironment->remove_BrowserProcessExited(m_browserExitedEventToken));
+              // Release the environment only after the handler is invoked.
```

Comment:
If multiple apps are sharing a user data folder, and the first app shuts down its WebView, the first app will never receive a BrowserProcessExited (since the browser process is still being used by other apps). How does it know when it's safe to proceed with shutdown? Is it the apps' responsibility to coordinate among themselves to keep track of "Will the last one out the door please turn off the lights (i.e., wait for BrowserProcessExited)?"

Reply:
Yes. The app can either coordinate with other processes sharing the user data folder or set a timer.

Reply:
Do we already note this in the documentation? If not please do.

Reply:
We had something about app processes coordination but not explicitly for waits, I added a comment about that.